Rock/Field/Types/TextFieldType.cs

```diff
@@ -74,10 +81,19 @@ public override List<Control> ConfigurationControls()
         {
             Dictionary<string, ConfigurationValue> configurationValues = new Dictionary<string, ConfigurationValue>();
             configurationValues.Add( IS_PASSWORD_KEY, new ConfigurationValue( "Password Field", "When set, edit field will be masked.", "" ) );
+            configurationValues.Add( MAX_LENGTH, new ConfigurationValue( "Maximum Length", "Enter the maximum length of this field.", "" ) );
 
-            if ( controls != null && controls.Count > 0 && controls[0] != null && controls[0] is CheckBox )
-            { 
-                configurationValues[IS_PASSWORD_KEY].Value = ( (CheckBox)controls[0] ).Checked.ToString();
+            if ( controls != null && controls.Count > 0 )
+            {
+                if ( controls[0] != null && controls[0] is CheckBox )
+                {
+                    configurationValues[IS_PASSWORD_KEY].Value = ( (CheckBox)controls[0] ).Checked.ToString();
+                }
+
+                if ( controls[1] != null && controls[1] is TextBox )
```

Comment:
This should also check to make sure that controls.Count > 1

Reply:
That was already checked at L86